lib/working_hours/computation.rb

```diff
@@ -80,7 +80,7 @@ def advance_to_working_time time, config: nil
         end
         # find first working range after time
         time_in_day = time.seconds_since_midnight
-        config[:working_hours][time.wday].each do |from, to|
+        (config[:holiday_hours][time.to_date] || config[:working_hours][time.wday]).each do |from, to|
```

Comment:
As you're doing this change in several places and some are still missing (`add_seconds`) IMO we should refactor this into a small method (something like `working_hours_for(time)`)
```suggestion
        working_hours_for(time).each do |from, to|
```
This would prevent any future refactoring from forgetting to update some places (like you did here so it's worth it ☺) and could also make testing easier if we want to add specs on this method (not require for now I would say).

Reply:
Fixed